S2VX.Game/Editor/Containers/NotesTimeline.cs

```diff
@@ -259,21 +259,43 @@ public class NotesTimeline : CompositeDrawable {
         }
 
         private void AddVisibleNotes() {
-            var lowerBound = Time.Current - SectionLength * SecondsToMS / 2;
-            var upperBound = Time.Current + SectionLength * SecondsToMS / 2;
+            var sectionDuration = SectionLength * SecondsToMS;
+            var lowerBound = Time.Current - sectionDuration / 2;
+            var upperBound = Time.Current + sectionDuration / 2;
             var color = Story.Notes.Colour;
             foreach (var note in Story.Notes.Children) {
-                if (lowerBound <= note.EndTime && note.EndTime <= upperBound) {
-                    var relativePosition = (note.EndTime - lowerBound) / (SectionLength * SecondsToMS);
-                    var visibleNote = new RelativeBox {
-                        Colour = color,
-                        Alpha = 0.727f,
-                        Width = TimelineNoteWidth,
-                        Height = TimelineNoteHeight,
-                        X = (float)relativePosition,
-                        Y = 0.2f,
-                        Anchor = Anchor.TopLeft,
-                    };
+                RelativeBox visibleNote = null;
+                var relativePosition = (note.HitTime - lowerBound) / sectionDuration;
+
+                if (note is EditorHoldNote holdNote) {
+                    if (lowerBound <= holdNote.HitTime || holdNote.EndTime <= upperBound) {
```

Comment:
The HoldNote getting stuck on the left side of NotesTimeline probably has something to do with the logic here. I tried changing the || to && but the HoldNote disappears too quickly Lol

Reply:
Fixed